public/app/features/alerting/unified/rule-editor/ExistingRuleEditor.tsx

```diff
@@ -1,23 +1,29 @@
 import { Alert, LoadingPlaceholder } from '@grafana/ui';
+import { useQueryParams } from 'app/core/hooks/useQueryParams';
 import { RuleIdentifier } from 'app/types/unified-alerting';
 
 import { AlertWarning } from '../AlertWarning';
 import { AlertRuleForm } from '../components/rule-editor/alert-rule-form/AlertRuleForm';
 import { useRuleWithLocation } from '../hooks/useCombinedRule';
 import { useIsRuleEditable } from '../hooks/useIsRuleEditable';
+import { RuleFormValues } from '../types/rule-form';
 import { stringifyErrorLike } from '../utils/misc';
 import * as ruleId from '../utils/rule-id';
 
 interface ExistingRuleEditorProps {
   identifier: RuleIdentifier;
+  /** Provide prefill if we are trying to restore an old version of an alert rule but we need the user to manually tweak the values */
+  prefill?: Partial<RuleFormValues>;
 }
 
-export function ExistingRuleEditor({ identifier }: ExistingRuleEditorProps) {
+export function ExistingRuleEditor({ identifier, prefill }: ExistingRuleEditorProps) {
```

Comment:
Nothing blocking but it feels like we try to push restoring logic to the `ExistingRuleEditor` and `AlertRuleForm`

How about having a separate `RestoreRuleEditor` component which merges data from existing rule and restored version and then pushes it to the form through, for example, a new `restoreFrom` prop?

I'm concerned about the complexity we introduce. We introduce interdependencies between prefill and existing which leads to much more cases we need to consider (is prefill, is existing, is prefill && existing)